### DISCUSSION TITLE
« Couldn't register hook »  : « Unexpected token 'export' »

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#10871] [Archive]
Directus 9.4.2 : I place the hook extracted from the [Directus documentation](https://docs.directus.io/extensions/hooks/#extension-entrypoint) in `my-directus-project/extensions/hooks/myfirsthook/index.js`

```js
// my-directus-project/extensions/hooks/myfirsthook/index.js

export default ({ filter, action }) => {
	filter('items.create', () => {
		console.log('Creating Item!');
	});

	action('items.create', () => {
		console.log('Item created!');
	});
};
```
When relaunching the Directus server, I get
```
Couldn't register hook "myfirsthook"
Unexpected token 'export'
```
How to fix that and get the hook properly registered?

**Comment by azrikahar** [comment]
Hi, are you using directus-extension to create the custom hook? The examples are assuming you are using it to build the output file in CommonJS syntax, which would be `module.exports` instead of `export default` for example. You can learn more about this from this comment here: https://github.com/directus/directus/discussions/10656#discussioncomment-1863343

**Reply by Pachat** [reply]
Thanks. And indeed, now the extension is loaded with
```
module.exports = ({ filter, action }) => {
	filter('items.create', () => {
		console.log('Creating Item!');
	});

	action('items.create', () => {
		console.log('Item created!');
	});
};
```
But adding an item in a collection, does not show `'Creating Item!'` neither `'Item created!'` in the console

**Reply by juldrig** [reply]
Here too, very confused

**Reply by pdaether** [reply]
You should include the collection name in the event name, like:
```javascript
filter('collection.items.create', myHookFn);
```

**Reply by azrikahar** [reply]
@Pachat seems to be the same as mentioned here: https://github.com/directus/directus/discussions/10832#discussioncomment-1930669. Hope that helps to resolve it 👍

**Reply by juldrig** [reply]
It worked guys, thx!
